Dockerfile-proxy

```diff
@@ -16,13 +16,13 @@ COPY pkg/tls pkg/tls
 COPY pkg/version pkg/version
 RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly ./pkg/...
 COPY proxy-identity proxy-identity
-RUN CGO_ENABLED=0 GOOS=linux go install -mod=readonly ./proxy-identity
+RUN CGO_ENABLED=0 GOOS=linux go build -o /out/proxy-identity -mod=readonly -ldflags "-s -w" ./proxy-identity
```

Comment:
Do we really want to strip all of the debug symbols out of the binaries? I'm not opposed to it but it seems like this could potentially make debugging (stacktraces etc) much more difficult?

Reply:
we don't lose stack traces with this change (which is something i agree we should not give up):

https://golang.org/cmd/link/
```
-s
	Omit the symbol table and debug information.
-w
	Omit the DWARF symbol table.
```

Reply:
My concern is that if we omit the symbol tables, the stack traces will be devoid of relevant information (like function/file names).

Reply:
Confirmed that with `-s -w` enabled:

```diff
diff --git a/bin/go-run b/bin/go-run
index 4f6704d9..0210d3b3 100755
--- a/bin/go-run
+++ b/bin/go-run
@@ -10,7 +10,7 @@ if [ "$#" -eq 0 ]; then
   exit 1
 fi

-ldflags="-X github.com/linkerd/linkerd2/pkg/version.Version=$($bindir/root-tag)"
+ldflags="-s -w -X github.com/linkerd/linkerd2/pkg/version.Version=$($bindir/root-tag)"
 GO111MODULE=on go build -v -mod=readonly -race -o .gorun -ldflags "$ldflags" "./$1"
 shift
 exec ./.gorun "$@"
diff --git a/cli/cmd/version.go b/cli/cmd/version.go
index 49097bb3..728dd20e 100644
--- a/cli/cmd/version.go
+++ b/cli/cmd/version.go
@@ -42,6 +42,9 @@ func newCmdVersion() *cobra.Command {
        cmd.PersistentFlags().BoolVar(&options.shortVersion, "short", options.shortVersion, "Print the version number(s) only, with no additional output")
        cmd.PersistentFlags().BoolVar(&options.onlyClientVersion, "client", options.onlyClientVersion, "Print the client version only")

+       var p *int
+       fmt.Printf("+%v", *p)
+
        return cmd
 }
```

...  stack trace is preserved:

```bash
$ bin/go-run cli version
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x29034a2]

goroutine 1 [running]:
github.com/linkerd/linkerd2/cli/cmd.newCmdVersion(0x3f65e40)
	/Users/sig/code/linkerd2/cli/cmd/version.go:46 +0x322
github.com/linkerd/linkerd2/cli/cmd.init.0()
	/Users/sig/code/linkerd2/cli/cmd/root.go:119 +0x960
```